content/concepts/security-considerations.md

```diff
@@ -0,0 +1,140 @@
+---
+title: 'Security Considerations'
+weight: 12
+---
+
+libp2p makes it simple to establish [encrypted, authenticated communication
+channels](../secure-comms/) between two peers, but there are other important
+security issues to consider when building robust peer-to-peer systems.
+
+Many of the issues described here have no known "perfect solution," and the
+solutions and mitigation strategies that do exist may come with tradeoffs and
+compromises in other areas. As a general-purpose framework, libp2p tries to
+provide the tools for application developers to address these problems, rather
+than taking arbitrary approaches to security that may not be acceptable to all
+systems built with libp2p.
+
+
+## Identity and Trust
+
+Every libp2p peer is uniquely identified by their [peer id](../peer-id/), which
+is derived from a private cryptographic key. Peer ids and their corresponding
+keys allow us to _authenticate_ remote peers, so that we can be sure we're
+talking to the correct peer and not an imposter.
+
+However, authentication is generally only half of the "auth" story when it comes
+to security. Many systems will also require _authorization_, or the ability to
+determine "who is allowed to do what."
+
+libp2p does not provide an authorization framework "out of the box", since the
+requirements vary widely across peer-to-peer systems. For example, some networks
+may not need authorization at all and can simply accept requests from any peer,
+while others may need to explicitly grant fine-grained permissions based on a
+hierarchy of roles, the requested resources or services, etc.
+
+To design an authorization system on libp2p, you can rely on the authentication
+of peer ids and build an association between peer ids and permissions, with the
+peer id serving the same function as the "username" in traditional authorization
+frameworks, and the peer's private key serving as the "password". Your [protocol
+handler](../protocols/) could then reject requests from untrusted peers.
+
+Of course, it's also possible to build other kinds of authorization systems on
+libp2p that are not based on peer ids. For example, you may want a single libp2p
+peer to be usable by many human operators, each with a traditional username and
+password. This could be accomplished by defining an authorization protocol that
+accepts usernames and passwords and responds with a signed token if the
+credentials are valid. Protocols that expose sensitive resources could then
+require the token before allowing access.
+
+Systems that are designed to be fully decentralized are often "open by default,"
+allowing any peer to participate in core functions. However, such systems may
+benefit from maintaining some kind "reputation" system to identify faulty or
```

Comment:
I think you are referring to a network-wide reputation system where peer scores are replicated and there's global consensus. However, local peer reputation can go a long way.